implementations/rust/ockam/ockam_api/src/kafka/outlet_service/prefix_forwarder.rs

```diff
@@ -0,0 +1,107 @@
+use crate::kafka::KAFKA_OUTLET_CONSUMERS;
+
+use crate::DefaultAddress;
+use core::str::from_utf8;
+use ockam::{Context, Result, Routed, Worker};
+use ockam_core::errcode::{Kind, Origin};
+use ockam_core::flow_control::{FlowControlId, FlowControlPolicy};
+use ockam_core::{Address, AllowAll, AllowOnwardAddress};
+
+/// This service applies a prefix to the provided static forwarding address.
+/// This service was created mainly to keep full compatibility with the existing
+/// erlang implementation.
+pub struct PrefixForwarderService {
```

Comment:
Could you please elaborate on what this service is doing and what is the use-case? Do you think it could be worth updating the original ForwardingService instead?

Reply:
The use case in the Erlang implementation is to avoid having a client registering arbitrary addresses inside the node, reducing the possibility of vulnerabilities.
It could be useful in this security sense only if we blocked the forwarding service by default.
By having this standalone service, we can have the same semantic of the orchestrator without having to adjust for special cases.

Reply:
It could be useful to generalize for keep compatibility reasons

Reply:
But what if the original ForwardingService had a prefix (could be empty) argument in the constructor? Would it then cover both cases?

Reply:
My main point is that writing a proper ForwardingService could be tricky, given AccessControls challenges. Having 2 implementations reduces our changes that both are correct :D